server/src/main/java/io/crate/types/StringType.java

```diff
@@ -78,7 +80,10 @@ public class StringType extends DataType<String> implements Streamer<String> {
         new EqQuery<Object>() {
 
             @Override
-            public Query termQuery(String field, Object value) {
+            public Query exactQuery(String field, Object value, boolean hasDocValues, IndexType indexType) {
+                if (indexType == IndexType.PLAIN && hasDocValues) {
```

Comment:
If there is index and docvalues, whey `newSlowRangeQuery`?

Reply:
`newSlowRangeQuery` utilizes doc-values which is only available if a text type is indexed with `StringIndexer`(`PLAIN`). If indexed with `FullTextIndexer`, `TermQuery` will be used.